SIMULACRO2.cpp

```diff
@@ -0,0 +1,376 @@
+#include <iostream>
+#include <string>
+
+using namespace std;
+
+class Vehiculo {
+protected:
+    string marca;
+    int velocidadMaxima;
+    int nivelCombustible;
+    int velocidadActual;
+
+public:
+    Vehiculo(string marcaNueva, int nuevaVelocidadMaxima, int nuevoNivelCombustible)
+        : marca(marcaNueva), velocidadMaxima(nuevaVelocidadMaxima), nivelCombustible(nuevoNivelCombustible), velocidadActual(0) {}
+
+    virtual void acelerar() = 0;
+
+    void frenar() {
+        if (velocidadActual > 0) {
+            velocidadActual -= 10;
+        }
+        if (velocidadActual < 0) {
+            velocidadActual = 0;
+        }
+    }
```

Comment:
Asi es mas directo y corto el codigo.
```suggestion
    void frenar() {
        if (velocidadActual > 0) {
            velocidadActual -= 10;
        } else {
            velocidadActual = 0;
        }
    }
```